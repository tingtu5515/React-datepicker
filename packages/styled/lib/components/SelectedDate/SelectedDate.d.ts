export interface SelectDateProps {
  title: string
  date: string
  isActive: boolean
}
declare function SelectDate({title, isActive, date}: SelectDateProps): JSX.Element
export default SelectDate
