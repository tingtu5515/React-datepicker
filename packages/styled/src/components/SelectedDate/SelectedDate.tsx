import React from 'react'
import styled, {css} from 'styled-components'
import useThemeProps from '../../hooks/useThemeProps'
// eslint-disable-next-line import/no-unresolved
import {SelectDateTheme} from '../../@types/theme'
import globalStyles from '../../globalStyles'
import Text from '../Text'

interface StyledDateProps {
  isActive: boolean
}
const StyledDate = styled(Text)<StyledDateProps>`
  position: relative;
  display: inline-block;

  &:after {
    content: '';
    position: absolute;
    height: 2px;
    width: 100%;
    bottom: 0;
    left: 0;
    z-index: 1;
  }

  ${({isActive}) =>
    isActive &&
    css`
      &:after {
        background: #00aeef;
      }
    `}
`

export interface SelectDateProps {
  title: string
  date: string
  isActive: boolean
}

function SelectDate({title, isActive, date}: SelectDateProps) {
  const theme: SelectDateTheme = useThemeProps({
    fontFamily: globalStyles.fontFamily,
    selectDateLabelFontSize: '11px',
    selectDateLabelColor: globalStyles.colors.silverCloud,
    selectDateLabelMargin: '0 0 8px',
    selectDateDateColor: globalStyles.colors.charcoal,
    selectDateDateFontSize: '24px',
    selectDateDateFontWeight: 500,
    selectDateDatePadding: '0 0 15px',
  })

  return (
    <div>
      <Text
        fontFamily={theme.fontFamily}
        fontSize={theme.selectDateLabelFontSize}
        // @ts-ignore
        color={theme.selectDateLabelColor}
        m={theme.selectDateLabelMargin}
      >
        {title}
      </Text>
      <StyledDate
        as="span"
        // @ts-ignore
        color={theme.selectDateDateColor}
        fontSize={theme.selectDateDateFontSize}
        fontWeight={theme.selectDateDateFontWeight}
        fontFamily={theme.fontFamily}
        p={theme.selectDateDatePadding}
        isActive={isActive}
      >
        {date}
      </StyledDate>
    </div>
  )
}

export default SelectDate
