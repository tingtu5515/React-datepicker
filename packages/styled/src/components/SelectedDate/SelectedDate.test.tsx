import * as React from 'react'
import {render} from 'react-testing-library'
import SelectedDate from '.'

test('should be unactive', () => {
  const {container} = render(
    <SelectedDate vertical={false} title="test" isActive={false} date="15.10.2018" />,
  )
  expect(container).toMatchSnapshot()
})

test('should be active', () => {
  const {container} = render(
    <SelectedDate isActive vertical={false} title="test" date="15.10.2018" />,
  )
  expect(container).toMatchSnapshot()
})

test('should render vertical variant', () => {
  const {container} = render(<SelectedDate vertical isActive title="test" date="15.10.2018" />)
  expect(container).toMatchSnapshot()
})
