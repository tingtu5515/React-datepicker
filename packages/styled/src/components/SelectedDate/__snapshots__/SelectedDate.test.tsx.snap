// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be active 1`] = `
.c0 {
  font-family: Montserrat,sans-serif;
  font-size: 11px;
  color: #929598;
  margin: 0 0 8px;
}

.c1 {
  font-family: Montserrat,sans-serif;
  font-size: 24px;
  font-weight: 500;
  color: #001217;
  padding: 0 0 15px;
  position: relative;
  display: inline-block;
}

.c1:after {
  content: '';
  position: absolute;
  height: 2px;
  width: 100%;
  bottom: 0;
  left: 0;
  z-index: 1;
}

.c1:after {
  background: #00aeef;
}

<div>
  <div>
    <div
      class="c0"
      color="#929598"
      font-family="Montserrat, sans-serif"
      font-size="11px"
    >
      test
    </div>
    <span
      class="c1"
      color="#001217"
      font-family="Montserrat, sans-serif"
      font-size="24px"
      font-weight="500"
    >
      15.10.2018
    </span>
  </div>
</div>
`;

exports[`should be unactive 1`] = `
.c0 {
  font-family: Montserrat,sans-serif;
  font-size: 11px;
  color: #929598;
  margin: 0 0 8px;
}

.c1 {
  font-family: Montserrat,sans-serif;
  font-size: 24px;
  font-weight: 500;
  color: #001217;
  padding: 0 0 15px;
  position: relative;
  display: inline-block;
}

.c1:after {
  content: '';
  position: absolute;
  height: 2px;
  width: 100%;
  bottom: 0;
  left: 0;
  z-index: 1;
}

<div>
  <div>
    <div
      class="c0"
      color="#929598"
      font-family="Montserrat, sans-serif"
      font-size="11px"
    >
      test
    </div>
    <span
      class="c1"
      color="#001217"
      font-family="Montserrat, sans-serif"
      font-size="24px"
      font-weight="500"
    >
      15.10.2018
    </span>
  </div>
</div>
`;
